cidrgrep.go

```diff
@@ -0,0 +1,31 @@
+package cidrgrep
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"regexp"
+)
+
+// Filter filters input for lines containing an IP within cidr
+func Filter(input io.Reader, output io.Writer, cidr string) {
+	scanner := bufio.NewScanner(input)
+
+	_, ipv4Net, err := net.ParseCIDR(cidr)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	re := regexp.MustCompile(`[0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}\.[0-9]{0,3}`)
```

Comment:
Regexp compilation is very expensive in comparison to actually matching against the compiled regexp. So when we use regexps in Go, which is nowhere near as often as in most other languages, we tend to compile them outside the function, at package level, and then refer to that compiled regexp in the function:

```go
var IPAddressRE = regexp.MustCompile(...)
...
func ... {
	if IPAddressRE.FindAllString....
```

Reply:
This is good to know. I have made this change too :-)